tools: add interactive docs-as-code tools matrix (#2684)
writethedocs/www: docs/guide/tools/docs-as-code-tools-matrix.rst

.. _docs-as-code-tools-matrix:

Documentation Tools Matrix
==========================

> A community-curated comparison of 27 documentation tools used in docs-as-code workflows,

Reviewer: Is there a need to hardcode the number of tools in the introduction?

> covering static site generators, hosted platforms, API doc tools, AI writing assistants, and more.

Reviewer: What are the AI writing assistants? If that should be covered here, it seems like it would have to be a lot more. Not sure it fits the focus.

.. note::

   **Last reviewed: May 2026.** This page is community-maintained — entries may become outdated as tools evolve.

>    If you spot something incorrect or missing, use the guide below to edit this page and submit a pull request,

Reviewer: ```suggestion
   If you spot something incorrect or missing, use the guide on :doc:`how to edit this page <docs-as-code-tools-matrix-contributing>`
```

**How to use this table:**

- **Search** — the search box filters across all visible columns
- **Column filters** — every column has a multi-select filter; click to select one or more values
- **Columns button** — show or hide additional columns grouped by category (authoring, publishing, scale, security, AI, and more)
- :ref:`Glossary <tools-matrix-glossary>` — definitions for column values, categories, and abbreviations used in this table
- :doc:`How to edit this page <docs-as-code-tools-matrix-contributing>` — add a tool, update an entry, or modify columns

.. raw:: html

   <link rel="stylesheet" type="text/css" href="../../../_static/css/datatables/dataTables.dataTables.min.css">
   <link rel="stylesheet" type="text/css" href="../../../_static/css/datatables/buttons.dataTables.min.css">
   <style>
     table.datatable tbody td { vertical-align: top; }
     table.datatable tbody td:nth-child(3) {
       font-size: 0.85em;
       max-width: 300px;
     }
     div.dt-buttons { margin-bottom: 8px; }
   </style>

.. csv-table::
   :class: datatable
   :header-rows: 1
   :file: ../../_data/tools.csv

.. raw:: html

>    <script src="../../../_static/js/datatables/jquery.min.js"></script>

Reviewer: I would suggest we could use something more lightweight than jQuery. Where is it being used in the scripts? I do not see what I think of as the telltale `$`. If it's just for DataTables, (other than suggesting maybe finding a different library) it's possible to build that into the DataTables script itself rather than needing separate ones.

   <script src="../../../_static/js/datatables/dataTables.buttons.min.js"></script>
   <script src="../../../_static/js/datatables/buttons.colVis.min.js"></script>
   <script>
   function initToolsMatrix() {
     // Sphinx csv-table wraps every cell in a <p> tag — unwrap before DataTables reads the table
     document.querySelectorAll('table.datatable td, table.datatable th').forEach(function (cell) {
       var p = cell.querySelector('p');
       if (p && cell.childElementCount === 1) {
         while (p.firstChild) { cell.insertBefore(p.firstChild, p); }
         p.remove();
       }
     });

     // Default visible: Tool (0), Category (2), Description (3), Cost (4), Markup Language (8)
     // Hidden by default: URL (1), Business Model (5) ... Local Models Supported (32)
     new DataTable('table.datatable', {
       dom: 'Bfrtip',
       buttons: [{
         extend: 'colvis',
         text: 'Columns',
         columnText: function (dt, idx, title) {
           var prefixes = {
             9: 'Author: ', 10: 'Author: ', 11: 'Author: ', 12: 'Author: ', 13: 'Author: ',
             14: 'Pub: ', 15: 'Pub: ', 16: 'Pub: ', 17: 'Pub: ', 18: 'Pub: ',

>              19: 'Scale: ', 20: 'Scale: ', 21: 'Scale: ', 22: 'Scale: '

Reviewer: This does not seem easy to maintain.
